pkg/nad/builder.go

```diff
@@ -155,6 +155,26 @@ func (builder *Builder) Delete() error {
 	return nil
 }
 
+// Update renovates the existing NAD object with nad definition in builder.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating NetworkAttachmentDefinition %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+
+	builder.Definition.CreationTimestamp = metaV1.Time{}
+	builder.Definition.ResourceVersion = builder.Object.ResourceVersion
```

Comment:
Why do we want to keep resorceversion?

Reply:
Yes, I'm not sure but it looks like it's required

Reply:
We don't do that  in the node package
https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/nodes/node.go#L63